### PR TITLE
add compile step in the samples to generate zip files

### PR DESCRIPTION
**Pull request by gaoning777**
fixing #1865 

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1866)

### CODE REVIEW

File: samples/core/resource_ops/resourceop_basic.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 It is not a good practice to put password as a pipeline argument, since it will
 be visible on KFP UI.
 """
-
+import kfp
```
Comment (numerology) [inline samples/core/resource_ops/resourceop_basic.py]: nit: can you switch the order of following two imports. thx.

File: samples/core/volume_ops/volumeop.py [hunk 2]
```diff
@@ -36,7 +36,5 @@ def volumeop_basic(size):
         pvolumes={"/mnt": vop.volume}
     )
 
-
-if __name__ == "__main__":
-    import kfp.compiler as compiler
-    compiler.Compiler().compile(volumeop_basic, __file__ + ".tar.gz")
+if __name__ == '__main__':
```
Comment (numerology) [inline samples/core/volume_ops/volumeop.py]: nit: Actually for top layer component two blank lines would be better

**Comment by gaoning777** [comment]
> Left some nit comments:)
> Q: what's the advantage of .zip over .tar.gz?

The only reason is that AIHub only support zip

File: samples/core/resource_ops/resourceop_basic.py (hunk 1 above)
Comment (gaoning777) [reply samples/core/resource_ops/resourceop_basic.py]: NP

File: samples/core/volume_ops/volumeop.py (hunk 2 above)
Comment (gaoning777) [reply samples/core/volume_ops/volumeop.py]: Not sure I understand, leaving two blank lines above main?

**Pull request merged**